src/db-service.ts

```diff
@@ -0,0 +1,103 @@
+// global
+import { sql, DatabaseTransactionConnectionType as TrxHandler } from 'slonik';
+import { Action } from './interfaces/action';
+
+
+/**
+ * Database's first layer of abstraction for Actions
+ */
+export class ActionService {
+  
+  // the 'safe' way to dynamically generate the columns names:
+  private static allColumns = sql.join(
+    [
+      'id',
+      ['member_id', 'memberId'],
+      ['member_type', 'memberType'],
+      ['item_id', 'itemId'],
+      ['item_type', 'itemType'],
+      ['action_type', 'actionType'],
+      'view',
+      'extra',
+      ['created_at', 'createdAt'],
+    ].map((c) =>
+      !Array.isArray(c)
+        ? sql.identifier([c])
+        : sql.join(
+            c.map((cwa) => sql.identifier([cwa])),
+            sql` AS `,
+          ),
+    ),
+    sql`, `,
+  );
+
+  /**
+   * Create given action and return it.
+   * @param action Action to create
+   * @param transactionHandler Database transaction handler
+   */
+  async create(action: Action, transactionHandler: TrxHandler,): Promise<Action> {
+    return transactionHandler
+      .query<Action>(
+        sql`
+        INSERT INTO "action" (
+            "member_id",
+            "member_type",
+            "item_id",
+            "item_type",
+            "action_type",
+            "view",
+            "extra"
```

Comment:
It is not possible to use `allColumns` because it includes the createdAt field, and it is not necessary when creating an action (it is done by the database itself).